Controller.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Response;
+
+class Controller
+{
+    /**
+     * Send JSON Response
+     *
+     * @param array $data
+     * @param integer $code
+     * @return Response
+     */
+    protected function sendJsonResponse(array $data, int $code): Response
+    {
+        return response()->json([
+            "code" => $code,
+            "data" => $data
+        ], $code);
+    }
+
+    /**
+     * Send JSON Error Response
+     *
+     * @param string $message
+     * @param integer $code
+     * @return Response
+     */
+    protected function sendErrorResponse(string $message, int $code): Response

```

Comment:
Some could say `$code` could have a default value with 200 for a normal response and some 400ish for error response.

Reply:
so should i make just 1 type of response method then check if `code < 400` then send normal response else send error response?

Reply:
Both would be fine I think, if you split them you have more freedom to change them later on without many if statements